src/main/java/org/opensearch/neuralsearch/processor/util/DocumentClusterManager.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.util;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Arrays;
+
+public class DocumentClusterManager {
+
+    int totalDocCounts; // total number of documents within the index
+    int[] clusterDocCounts; // number of documents across each cluster
+    float[][] clusterRepresentatives; // an array of sketch vectors indicating the center of each cluster
+
+    // Resource paths relative to classpath
+    private static final int SKETCH_SIZE = 1024;
+    private static final int CLUSTER_NUM = 11896;
+    private static final String CLUSTER_ASSIGNMENT_RESOURCE = "assignment.bin";
+    private static final String CLUSTER_REPRESENTATIVE_RESOURCE = "representatives.bin";
+
+    // Instance is created at class loading time
+    private static final DocumentClusterManager INSTANCE = new DocumentClusterManager();
+
+    private DocumentClusterManager() {
+        // Private constructor due to singleton
+
+        loadClusterAssignment();
```

Comment:
I'd recommend to put these loadXX functions to a new function `initialize` so that you can have a control on when they should be loaded.